identity/src/test/java/com/android/identity/util/SimpleApplicationDataTest.java

```diff
@@ -0,0 +1,338 @@
+
+        appData.setData("bar", new byte[0]);
+        assertEquals(Timestamp.ofEpochMilli(0), appData.getTimestamp("foo"));
+        assertThrows(IllegalArgumentException.class, () -> appData.getTimestamp("bar"));
+
+        appData.setString("bar", "abc");
+        assertEquals(Timestamp.ofEpochMilli(0), appData.getTimestamp("foo"));
+        assertThrows(IllegalArgumentException.class, () -> appData.getTimestamp("bar"));
+
+        appData.setNumber("bar", 82L);
+        assertEquals(Timestamp.ofEpochMilli(0), appData.getTimestamp("foo"));
+        assertEquals(Timestamp.ofEpochMilli(82), appData.getTimestamp("bar"));
+
+        appData.setBoolean("bar", true);
+        assertEquals(Timestamp.ofEpochMilli(0), appData.getTimestamp("foo"));
+        assertThrows(IllegalArgumentException.class, () -> appData.getTimestamp("bar"));
+    }
+
+    @Test
+    public void testBooleanValues() {
+        SimpleApplicationData appData = new SimpleApplicationData();
+
+        appData.setBoolean("foo", true);
+        assertEquals(true, appData.getBoolean("foo"));
+        testEncodingConsistency(appData, SimpleApplicationData.decodeFromCbor(appData.encodeAsCbor()));
+
+        appData.setBoolean("bar", false);
+        assertEquals(true, appData.getBoolean("foo"));
+        assertEquals(false, appData.getBoolean("bar"));
+        testEncodingConsistency(appData, SimpleApplicationData.decodeFromCbor(appData.encodeAsCbor()));
+
+        // remove by nulling
+        appData.setData("bar", (byte[]) null);
+        assertEquals(true, appData.getBoolean("foo"));
+        testEncodingConsistency(appData, SimpleApplicationData.decodeFromCbor(appData.encodeAsCbor()));
+        assertFalse(appData.keyExists("bar"));
+        assertThrows(IllegalArgumentException.class, () -> appData.getString("bar"));
+
+        // non-boolean values being read as a boolean
+        appData.setData("bar", new byte[] {0x53, 0x54, 0x55});
+        assertEquals(true, appData.getBoolean("foo"));
+        assertThrows(IllegalArgumentException.class, () -> appData.getBoolean("bar"));
+
+        appData.setData("bar", new byte[0]);
+        assertEquals(true, appData.getBoolean("foo"));
+        assertThrows(IllegalArgumentException.class, () -> appData.getBoolean("bar"));
+        testEncodingConsistency(appData, SimpleApplicationData.decodeFromCbor(appData.encodeAsCbor()));
+
+        appData.setString("bar", "abc");
+        assertEquals(true, appData.getBoolean("foo"));
+        assertThrows(IllegalArgumentException.class, () -> appData.getBoolean("bar"));
+
+        appData.setNumber("bar", 82L);
+        assertEquals(true, appData.getBoolean("foo"));
+        assertThrows(IllegalArgumentException.class, () -> appData.getBoolean("bar"));
+
+        appData.setTimestamp("bar", Timestamp.ofEpochMilli(0));
+        assertEquals(true, appData.getBoolean("foo"));
+        assertThrows(IllegalArgumentException.class, () -> appData.getBoolean("bar"));
+    }
```

Comment:
There's a lot of tests for convenience getters/setters which is a bit pointless b/c they essentially are one- and two-line methods respectively. That is, for subtype `Baz` they look like this (with NonNull annotations omitted)

```
SimpleApplicationData setBaz(String key, Baz value) {
    setData(key, Util.cborEncodeBaz(value));
    return this;
}

Baz getBaz(String key) {
    return Util.cborDecodeBaz(getData(key));
}
```

This is not to say that these tests are not a good idea, they are... but I think it would be more helpful if they are tests for the `Util.cborDecodeBaz()` functions for X = {String, Number, Timestamp, Boolean}. Because if we have those guarantees I think it's fine to not add tests for something that is a simple one- or two-line method.

So would you mind moving the tests to `com.android.identity.internal.UtilTest` where we have all the tests for the `Util.cborDecodeX()` helpers?

Reply:
Following our conversation about this, I added the parts of the tests that specifically rely on `com.android.identity.internal.Util` to `com.android.identity.internal.UtilTest` and left the existing tests where they are :)